app/models/ems_refresh/save_inventory_network.rb

```diff
@@ -378,38 +376,43 @@ def save_load_balancer_health_checks_inventory(ems, hashes, target = nil)
               end
 
     hashes.each do |h|
-      %i(load_balancer load_balancer_listener).each do |relation|
-        h[relation] = h.fetch_path(relation, :_object) if h.fetch_path(relation, :_object)
-      end
+      h[:load_balancer_id] = h.fetch_path(:load_balancer, :id)
+      h[:load_balancer_listener_id] = h.fetch_path(:load_balancer_listener, :id)
     end
 
     save_inventory_multi(ems.load_balancer_health_checks,
                          hashes,
                          deletes,
                          [:ems_ref],
-                         :load_balancer_health_check_members)
+                         :load_balancer_health_check_members,
+                         [:load_balancer, :load_balancer_listener])
     store_ids_for_new_records(ems.load_balancer_health_checks, hashes, :ems_ref)
   end
 
   def save_load_balancer_health_check_members_inventory(load_balancer_health_check, hashes)
     deletes = load_balancer_health_check.load_balancer_health_check_members.reload.dup
 
     hashes.each do |h|
-      %i(load_balancer_pool_member).each do |relation|
-        h[relation] = h.fetch_path(relation, :_object) if h.fetch_path(relation, :_object)
-      end
+      h[:load_balancer_pool_member_id] = h.fetch_path(:load_balancer_pool_member, :id)
     end
 
-    save_inventory_multi(load_balancer_health_check.load_balancer_health_check_members, hashes, deletes,
+    save_inventory_multi(load_balancer_health_check.load_balancer_health_check_members,
+                         hashes,
+                         deletes,
+                         [:load_balancer_pool_member_id],
+                         nil,
                          [:load_balancer_pool_member])
   end
 
   def link_cloud_subnets_to_network_routers(hashes)
     return if hashes.blank?
 
+    cloud_subnets = CloudSubnet.where(:id => hashes.map { |x| x[:id] }.compact.uniq).find_each.index_by(&:id)
```

Comment:
Would it make sense to skip active record and just hit the db on this?
Not sure if this would sidestep some validations or change expectations

```ruby
CloudSubnet.where(:id => hash[:is]).update_all(:network_router_id => network_router)
```
```